src/twisted/application/test/test_internet.py

```diff
@@ -1031,3 +1031,78 @@ def test_stopServiceOnStoppedService(self):
 
         self.assertIsNone(self.successResultOf(firstStopDeferred))
         self.assertIsNone(self.successResultOf(secondStopDeferred))
+
+
+    def test_onNewConnectionCalledWhenServiceStarts(self):
+        """
+        The C{onNewConnection} callable is called when the
+        L{ClientService.startService} is called.
+        """
+        newConnections = [0]
+        def onNewConnection(_proto):
+            newConnections[0] += 1
+
+        cq, service = self.makeReconnector(onNewConnection=onNewConnection)
```

Comment:
Since the docstring talks about startService,
either use

```
cq, service = self.makeReconnector(onNewConnection=onNewConnection, startService=True)
```

or better, explicit about  what is going on here

```
clock = Clock()
cq, service = self.makeReconnector(
    onNewConnection=onNewConnection,
    startService=False,
    fireImmediatly=False,
    clock=clock)

service.startService()

# The initial connection fails.
cq.connectQueue[0].errback(Exception('failed-connection'))

# Not yet called.
self.assertEqual(0, newConnections[0])

# The connection is made on the retry.
clock.advance(AT_LEAST_ONE_ATTEMPT)
cq.connectQueue[1].callback(None)

self.assertEqual(1, newConnections[0])
```